projects/storefrontlib/src/cms-components/asm/services/asm-enabler.service.ts

```diff
@@ -77,13 +71,6 @@ export class AsmEnablerService {
    * Adds the ASM UI by using the `cx-storefront` outlet.
    */
   private addUi(): void {
-    if (this.isUiAdded) {
-      return;
-    }
-    const factory = this.componentFactoryResolver.resolveComponentFactory(
-      AsmMainUiComponent
-    );
-    this.triggerService.renderDialog(this.triggerConfig.trigger?.asm, factory);
-    this.isUiAdded = true;
+    this.triggerService.renderDialog(TRIGGER_CALLER.ASM, AsmMainUiComponent);
```

Comment:
really like it now. What about renaming to "render"? We could potentially redirect to a page, which isn't really a dialog anymore.